packages/vite/src/node/config.ts

```diff
@@ -1034,6 +1034,16 @@ function resolveDepOptimizationOptions(
   )
 }
 
+export function isResolvedConfig(
+  inlineConfig: InlineConfig | ResolvedConfig,
+): inlineConfig is ResolvedConfig {
+  // assume that internal methods cannot be provided in inline config
+  return (
+    'fsDenyGlob' in inlineConfig &&
+    typeof inlineConfig.fsDenyGlob === 'function'
+  )
+}
```

Comment:
we discussed in the last meeting that it is better to have a symbol to identify the resolved config